### PR TITLE
Harmonize Gnome3 Defaults

### LINKED ISSUE #67310: Cleanup GNOME3 default applications
**Describe the bug**
Currently, the gnome3 module automatically adds the following to `environment.systemPackages`

```
optionalPackages = with gnome3; [ baobab eog epiphany evince
  gucharmap nautilus totem vino yelp gnome-bluetooth
  gnome-calculator gnome-contacts gnome-font-viewer gnome-screenshot
  gnome-system-monitor simple-scan
  gnome-terminal gnome-user-docs evolution file-roller gedit
  gnome-clocks gnome-music gnome-tweaks gnome-photos
  nautilus-sendto dconf-editor vinagre gnome-weather gnome-logs
  gnome-maps gnome-characters gnome-calendar accerciser gnome-nettool
  gnome-getting-started-docs gnome-packagekit gnome-software
  gnome-power-manager gnome-todo pkgs.gnome-usage
];
```

I know that `gnome-usage` is still experimental, and probably shouldn't be installed by default.

Does GNOME3 document their distribution defaults anywhere?
I think we could clean this up slightly.

Note that if we remove anything from here it would be a good idea to conditionally add them based off the users `stateVersion`. That way current users won't see these packages magically disappear from their machine.

Or perhaps it's better document that in the release notes.

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
See commit messages for what was removed and added.

Fixes https://github.com/NixOS/nixpkgs/issues/67310

###### Things to consider in this PR

It seems upstream has moved to Evolution to World so it isn't in core-utilities. We're missing a default email application.

We could either re-add it or swap it for Geary. I'd recommend the latter, and it seems to already be planned
* https://gitlab.gnome.org/GNOME/gnome-build-meta/issues/144

Do we want to remove telepathy from the defaults?
It's runtime optional in gnome-shell, though I think for this to work in NixOS we'd have to use `sessionPath` but it would be limited to the `telepathy` module. We'd then have to disable the functionality in the default apps.

I don't think we should include `gnome-boxes` default. I'm pretty sure it'd bring in some pretty large dependencies, I don't think a default experience needs a virtual machine application.
Though we do need something in the future for remote desktop viewing.

Add `seahorse` default. It was recently added to world
* https://gitlab.gnome.org/GNOME/gnome-build-meta/merge_requests/224

My opinion is that it's a useful application for secrets which are already managed default.

Remove vino
* https://gitlab.gnome.org/GNOME/gnome-build-meta/commit/c802b89a1edc14d4725f89d6b88d38fc609629bc

This might be something to do later, I'm not sure if gnome-shell has it runtime optional.

---

Release notes will be prepared once approval is given.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jtojnar** [comment]
I would vote for disabling penultimate by default on NixOS. It is non-standard package causing significant deviations from upstream. If someone wants to have it, they can always enable it.

File: nixos/modules/services/x11/desktop-managers/gnome3.nix [hunk 1]
```diff
@@ -281,23 +286,22 @@ in
       ];
     })
 
+    # Adapt from https://gitlab.gnome.org/GNOME/gnome-build-meta/blob/gnome-3-32/elements/core/meta-gnome-core-utilities.bst
     (mkIf serviceCfg.core-utilities.enable {
       environment.systemPackages = (with pkgs.gnome3; removePackagesByName [
-        baobab eog epiphany evince gucharmap nautilus totem yelp gnome-calculator
```
Comment (jtojnar) [inline nixos/modules/services/x11/desktop-managers/gnome3.nix]: Can we make this one per line so the diff is easier analyze?

**Comment by jtojnar** [comment]
> It seems upstream has moved to Evolution to World so it isn't in core-utilities. We're missing a default email application.

Do we need one? Lot of people use web mails these days.

> We could either re-add it or swap it for Geary. I'd recommend the latter, and it seems to already be planned

Works for me, as that is what I use.

> Do we want to remove telepathy from the defaults?
> It's runtime optional in gnome-shell, though I think for this to work in NixOS we'd have to use `sessionPath` but it would be limited to the `telepathy` module. We'd then have to disable the functionality in the default apps.

I would rather keep it for now (at least until they drop support). It would be in line with our custom of building packages with as wide feature set as possible. Though, if it is only a runtime dependency (via `GI_TYPELIB_PATH`) maybe it would be permissible.

You know my abhorrence of `sessionPath`, so the only alternative would be something like `gnome-shell-with-packages` and I am too lazy to implement that. The harder part would actually be the need to decouple gnome-shell from gnome-session and we already need that for [Shell-less Flashback](https://github.com/NixOS/nixpkgs/issues/47451#issuecomment-525066786) so maybe, it is not such a bad idea.

> I don't think we should include `gnome-boxes` default. I'm pretty sure it'd bring in some pretty large dependencies, I don't think a default experience needs a virtual machine application.

I agree with cutting out Boxes for being huge.

> Though we do need something in the future for remote desktop viewing.

Does that need to be in core-utils? It feels like a too specialist task to justify such a huge closure.

> Add `seahorse` default. It was recently added to world
> My opinion is that it's a useful application for secrets which are already managed default.

How often does one need it though? I run it maybe once a year when Polari somehow stops being able to use the stored passwords and I need to re-enter them manually.

> Remove vino

We need to keep it until https://gitlab.gnome.org/GNOME/gnome-control-center/issues/212 is resolved.

**Comment by jtojnar** [comment]
> > > Do we want to remove telepathy from the defaults?
> > > It's runtime optional in gnome-shell, though I think for this to work in NixOS we'd have to use `sessionPath` but it would be limited to the `telepathy` module. We'd then have to disable the functionality in the default apps.
> > 
> > 
> > I would rather keep it for now (at least until they drop support). It would be in line with our custom of building packages with as wide feature set as possible. Though, if it is only a runtime dependency (via `GI_TYPELIB_PATH`) maybe it would be permissible.
> 
> It does seem they've already dropped it from core
> 
>     * [gitlab.gnome.org/GNOME/gnome-build-meta/merge_requests/24](https://gitlab.gnome.org/GNOME/gnome-build-meta/merge_requests/24)

I was looking at it from NixOS perspective – that is dropping support from GNOME Shell code base. Every disabled feature means someone will probably need to rebuild the package with the feature enabled and will not be able to use the binary cache. At least with wrappers we would avoid GNOME Shell rebuild but , until https://github.com/NixOS/nixpkgs/issues/67546 is fixed, gnome-session will still be affected.

> Previously we shipped a remote desktop viewer "vinagre", not sure if there's an equivalent.

I am afraid GNOME decided it belongs into Boxes, so no separate app. Well, we do have Remmina but it is far from lightweight itself.

> Well I was thinking, it's a utility linked to `gnome-keyring` which is a service needed default.
> So general use of the tools the DE provides us. Think of it as a utility like `baobab` which may not get frequent use.

It is pretty small and unobtrusive so I am not against it.

File: nixos/modules/services/x11/desktop-managers/gnome3.nix [hunk 2]
```diff
@@ -257,20 +259,27 @@ in
       fonts.fonts = with pkgs; [
         cantarell-fonts
         dejavu_fonts
+        noto-fonts-emoji
```
Comment (jtojnar) [inline nixos/modules/services/x11/desktop-managers/gnome3.nix]: Could we make a `emojiFont` option until the [conflict](https://github.com/NixOS/nixpkgs/pull/67014#issuecomment-522839053) is resolved? I suspect it is caused by JoyPixels missing from, will test the hypothesis.

**Comment by worldofpeace** [comment]
Added release notes @jtojnar (maybe they could be better?). We should merge with, https://github.com/NixOS/nixpkgs/pull/67667.

File: nixos/doc/manual/release-notes/rl-1909.xml [hunk 3]
```diff
@@ -57,6 +57,64 @@
       and <option>services.xserver.desktopManager.xfce4-14</option> simultaneously or to downgrade from Xfce 4.14 after upgrading.
     </para>
    </listitem>
+   <listitem>
+    <para>
+      The GNOME 3 desktop manager module sports an interface to enable/disable core services, applications, and optional GNOME packages
+      like games.
+      <itemizedlist>
+      <para>This can be achieved with the following options which the desktop manager default enables</para>
+      <listitem><para><link linkend="opt-services.gnome3.core-os-services.enable"><literal>services.gnome3.core-os-services.enable</literal></link></para></listitem>
```
Comment (jtojnar) [inline nixos/doc/manual/release-notes/rl-1909.xml]: Cannot the `link` content be filled in automatically?

File: nixos/doc/manual/release-notes/rl-1909.xml [hunk 4]
```diff
@@ -57,6 +57,64 @@
       and <option>services.xserver.desktopManager.xfce4-14</option> simultaneously or to downgrade from Xfce 4.14 after upgrading.
     </para>
    </listitem>
+   <listitem>
+    <para>
+      The GNOME 3 desktop manager module sports an interface to enable/disable core services, applications, and optional GNOME packages
+      like games.
+      <itemizedlist>
+      <para>This can be achieved with the following options which the desktop manager default enables</para>
```
Comment (jtojnar) [inline nixos/doc/manual/release-notes/rl-1909.xml]: Games are not on by default, are they?

File: nixos/doc/manual/release-notes/rl-1909.xml [hunk 5]
```diff
@@ -57,6 +57,64 @@
...
+      </itemizedlist>
+      With these options we hope to give users finer grained control over their systems. Prior to this change you'd either have to manually
+      disable options or use <option>environment.gnome3.excludePackages</option> which only excluded the optional applications.
+      <option>environment.gnome3.excludePackages</option> is now unguarded, it can exclude any package installed with <option>environment.systemPackages</option>
+      in the GNOME 3 module.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Orthoganal to the previous changes to the GNOME 3 desktop manager module, we've updated all default services and applications
```
Comment (jtojnar) [inline nixos/doc/manual/release-notes/rl-1909.xml]: gonal

File: nixos/doc/manual/release-notes/rl-1909.xml (hunk 4 above)
Comment (worldofpeace) [reply nixos/doc/manual/release-notes/rl-1909.xml]: Oops, I'll express that they aren't

File: nixos/doc/manual/release-notes/rl-1909.xml (hunk 3 above)
Comment (worldofpeace) [reply nixos/doc/manual/release-notes/rl-1909.xml]: Doesn't appear so.

**Review by jtojnar — APPROVED** [review]

File: nixos/doc/manual/release-notes/rl-1909.xml (hunk 3 above)
Comment (jtojnar) [reply nixos/doc/manual/release-notes/rl-1909.xml]: Does not `<xref linkend="opt-services.gnome3.core-os-services.enable"/>`

Comment (worldofpeace) [reply nixos/doc/manual/release-notes/rl-1909.xml]: https://github.com/NixOS/nixpkgs/pull/68196

**Pull request merged**